artiq/gui/entries.py

```diff
@@ -52,10 +53,12 @@ def set_argument(self, key, argument):
         self._arguments[key] = argument
         widgets = dict()
         self._arg_to_widgets[key] = widgets
+        if argument["desc"].get("quickstyle") and not self._enable_quickstyle:
+            argument["desc"]["quickstyle"] = False
```

Comment:
To avoid this mutation, I made the following changes:

1. Make a copy of the `argument["desc"]` dictionary.
2. Separate out `quickstyle` functionality into a separate class and handle the logic in the `procdesc_to_entry` function instead of internally in `EnumerationEntry` (similar to `NumberEntryFloat` vs `NumberEntryInt`).